### ISSUE TITLE
Lookup Filter - Exclude values which were selected

### ISSUE THREAD

**Issue by cfthong** [volkanceylan/Serenity#300]
Let's take Customer's Order as an example. When you click on New Order Detail, the Product ID will list out all the products in the dropdownlist. When you tried to save a product which already exists in the order, the system will prompt the error message "This product is already in order details!",

Is there a way for us to remove the selected product from the dropdownlist?

Please advise.

**Comment by volkanceylan** [comment]
You need to write custom editor for this. Subclass lookup editor and override GetItems method. Anyway its users reponsibility to not add same product twice, as there is unique key in Northwind. If you hide product, they'll be confused more.

**Comment by cfthong** [comment]
Thanks for your prompt reply, but how do you pass the value (i.e. Order ID) to the lookup editor for it to do the filtering?

**Comment by volkanceylan** [comment]
Override CreateEntityDialog of grid editor, pass option to dialog. Dialog passes option to editor in its afterloadentity, lookup calls UpdateItems and filters in GetItems.

You should pass product ids to filter not order id.

**Comment by cfthong** [comment]
Why Product ID? I should filter out all those products which already selected in Order details.

For example if I tried to select new product for existing Order ID 10248:
select *
from products
where ProductID not in (select productID from [Order Details] where orderid = 10248)

if I tried to amend the existing product for existing Order ID 10248 and Detail ID 1109:
select *
from products
where ProductID not in (select productID from [Order Details] where orderid = 10248 and (DetailID <> 1109))

Not sure if I have explained what I wanted correctly.... In stored procedure, I will do something like this:
select *
from products
where ProductID not in (select productID from [Order Details] where orderid = @OrderID and (DetailID <> @DetailID or @DetailID = 0))

For new Order, i will pass in the following parameter @OrderID = 0 & @DetailID = 0;
For existing Order & new Order Detail, I will pass in following parameter @OrderID = Order ID & @DetailID = 0;
For existing Order & existing Order Detail, I will pass in the following parameter @OrderID = Order ID and @DetailID = Order Detail ID.
